### PR TITLE
DCOS-8087: hide docker volumes if no container image is provided

### PR DESCRIPTION
**Pull request by Poltergeist**
This is mostly @MatApple work, thanks again for jumping on this one. 
What it does is removing the docker volumes from the volumes section if no docker image is provided. 
docker-volumes mov

### CODE REVIEW

**Comment by mesosphere-web** [comment]
Reviewer: @orlandohohmeier

File: src/js/components/ServiceForm.js [hunk 1]
```diff
@@ -332,7 +361,8 @@ class ServiceForm extends SchemaForm {
           return port;
         });
       }
-      this.internalStorage_set({model});
+      let {dockerVolumesDefinition = null} = this.internalStorage_get();
```
Comment (philipnrmn) [inline src/js/components/ServiceForm.js]: 💯 destructuring defaults

**Comment by philipnrmn** [comment]
Nice!

File: src/js/components/ServiceForm.js [hunk 2]
```diff
@@ -233,8 +233,37 @@ class ServiceForm extends SchemaForm {
...
   updateDefinitions() {
     let {model} = this.internalStorage_get();
-    let {networking} = this.multipleDefinition;
+    let {networking, volumes} = this.multipleDefinition;
+    let {containerSettings} = model;
 
+    let showDockerVolumes = containerSettings &&
+      containerSettings.image != null && containerSettings.image !== '';
+
+    let endIndex = -1;
```
Comment (orlandohohmeier) [inline src/js/components/ServiceForm.js]: I really don't like that we rely on these indices, but I can't think of a better solution without refactoring huge parts of form. Let's make sure we get rid of these once we refactor the schema form.

**Comment by orlandohohmeier** [comment]
LGTM! 🚢

**Pull request merged**